datafusion/sql/tests/cases/diagnostic.rs

```diff
@@ -351,6 +351,36 @@ fn test_in_subquery_multiple_columns() -> Result<(), Box<dyn std::error::Error>>
             .collect::<Vec<_>>(),
         vec!["Select only one column in the subquery"]
     );
+    Ok(())
+}
+
+#[test]
+fn test_unary_op_plus_with_column() -> Result<()> {
+    // Test with a direct query that references a column with an incompatible type
+    let query = "SELECT +/*whole*/first_name/*whole*/ FROM person";
+    let spans = get_spans(query);
+    let diag = do_query(query);
+    assert_eq!(diag.message, "+ cannot be used with Utf8");
+    assert_eq!(diag.span, Some(spans["whole"]));
+    Ok(())
+}
 
+#[test]
+fn test_unary_op_plus_with_non_column() -> Result<()> {
+    // create a table with a column of type varchar
+    let query = "SELECT /*whole*/+'a'/*whole*/ ";
```

Comment:
I would remove the span tags `/*whole*/` since they aren't used by the test anyway